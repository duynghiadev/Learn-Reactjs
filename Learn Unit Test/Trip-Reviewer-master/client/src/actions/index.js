import axios from 'axios';

import {
  GET_TRIPS,
  GET_USERS,
  GET_TRIP_W_REVIEWER,
  CLEAR_TRIP_W_REVIEWER,
  GET_TRIP_W_REVIEWER_ERROR,
  USER_REGISTER,
  USER_LOGIN,
  RESET_PASSWORD,
  USER_AUTH,
  ADD_TRIP,
  UPDATE_TRIP,
  UPDATE_USER,
  DELETE_TRIP,
  GET_TRIP,
  GET_TRIP_ERROR,
  CLEAR_NEW_TRIP,
  CLEAR_TRIP,
  GET_USER_REVIEWS,
} from '../constants/action-names';

export function getTrips(limit = 10, start = 0, order = 'asc', list = []) {
  const request = axios.get(`/api/getManyTrips?limit=${limit}&skip=${start}&order=${order}`);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: GET_TRIPS,
        payload: {
          list: [...list, ...response.data],
          newTripsCount: response.data.length,
        },
      });
    });
}

export function getTripWithReviewer(id) {
  const request = axios.get(`/api/getTrip?id=${id}`);

  return dispatch =>
    request
      .then(({ data: tripData }) => {
        const trip = tripData;

        return axios.get(`/api/getReviewer?id=${trip.ownerId}`).then(({ data: reviewerData }) => {
          const response = {
            trip,
            reviewer: reviewerData,
          };

          dispatch({
            type: GET_TRIP_W_REVIEWER,
            payload: response,
          });
        });
      })
      .catch(error =>
        dispatch({
          type: GET_TRIP_W_REVIEWER_ERROR,
          payload: error,
        })
      );
}

export function getUserReviews(userId) {
  const request = axios.get(`/api/getUserReviews?user=${userId}`);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: GET_USER_REVIEWS,
        payload: response.data,
      });
    });
}

export function updateUser(data) {
  const request = axios.post('/api/userUpdate', data);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: UPDATE_USER,
        payload: response.data,
      });
    });
}

export function getTrip(id) {
  const request = axios.get(`/api/getTrip?id=${id}`);

  return dispatch =>
    request
      .then(response => {
        dispatch({
          type: GET_TRIP,
          payload: response.data,
        });
      })
      .catch(error => {
        dispatch({
          type: GET_TRIP_ERROR,
          payload: error,
        });
      });
}

export function updateTrip(data) {
  const request = axios.post('/api/tripUpdate', data);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: UPDATE_TRIP,
        payload: response.data,
      });
    });
}

export function deleteTrip(id) {
  const request = axios.delete(`/api/tripDelete?id=${id}`);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: DELETE_TRIP,
        payload: response.data,
      });
    });
}

export function clearTrip() {
  return {
    type: CLEAR_TRIP,
    payload: {
      trip: null,
      updatedTrip: false,
      postDeleted: false,
    },
  };
}

export function clearTripWithReviewer() {
  return {
    type: CLEAR_TRIP_W_REVIEWER,
    payload: {
      trip: null,
      reviewer: null,
    },
  };
}

export function loginUser({ email, password }) {
  const request = axios.post('/api/login', { email, password });

  return dispatch =>
    request.then(response => {
      dispatch({
        type: USER_LOGIN,
        payload: response.data,
      });
    });
}

export function resetPassword({ email }) {
  const request = axios.post('/api/resetPassword', { email });

  return dispatch =>
    request.then(response => {
      dispatch({
        type: RESET_PASSWORD,
        payload: response.data,
      });
    });
}

export function auth() {
  const request = axios.get('/api/auth');

  return dispatch =>
    request.then(response => {
      dispatch({
        type: USER_AUTH,
        payload: response.data,
      });
    });
}

export function addTrip(trip) {
  const request = axios.post('/api/trip', trip);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: ADD_TRIP,
        payload: response.data,
      });
    });
}

export function clearNewTrip() {
  return {
    type: CLEAR_NEW_TRIP,
    payload: null,
  };
}

export function getUsers() {
  const request = axios.get('/api/users');

  return dispatch =>
    request.then(response => {
      dispatch({
        type: GET_USERS,
        payload: response.data,
      });
    });
}

export function userRegister(user) {
  const request = axios.post('/api/register', user);

  return dispatch =>
    request.then(({ data }) => {
      dispatch({
        type: USER_REGISTER,
        payload: data,
      });
    });
}
