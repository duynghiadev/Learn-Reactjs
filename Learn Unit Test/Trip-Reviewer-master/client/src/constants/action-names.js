export const GET_TRIPS = 'get-trips';
export const GET_USERS = 'get-users';
export const ADD_TRIP = 'add-trip';
export const UPDATE_TRIP = 'update-trip';
export const UPDATE_USER = 'update-user';
export const DELETE_TRIP = 'delete-trip';
export const GET_TRIP = 'get-trip';
export const GET_TRIP_ERROR = 'get-trip-error';
export const GET_TRIP_W_REVIEWER = 'get-trip-w-reviewer';
export const GET_TRIP_W_REVIEWER_ERROR = 'get-trip-w-reviewer-error';
export const CLEAR_TRIP_W_REVIEWER = 'clear-trip-w-reviewer';
export const CLEAR_NEW_TRIP = 'clear-new-trip';
export const CLEAR_TRIP = 'clear-trip';
export const USER_REGISTER = 'user-register';
export const USER_LOGIN = 'user-login';
export const RESET_PASSWORD = 'reset-password';
export const USER_AUTH = 'user-auth';
export const GET_USER_REVIEWS = 'get-user-reviews';
