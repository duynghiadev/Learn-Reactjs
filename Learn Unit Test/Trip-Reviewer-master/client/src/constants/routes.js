export const HOME = '/';
export const LOGIN = '/login';
export const RESET = '/reset-password';
export const LOGOUT = '/user/logout';
export const REGISTER = '/user/register';
export const USER_PROFILE = '/user';
export const USER_REVIEWS = '/user/user-reviews';
export const EDIT_PROFILE = '/user/edit-profile';
export const EDIT_POST = '/user/edit-post';
export const ADD_TRIP = '/trip/add';
export const TRIPS = '/trips';
