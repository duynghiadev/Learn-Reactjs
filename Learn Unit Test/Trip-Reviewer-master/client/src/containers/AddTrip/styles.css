.add-review-container {
  text-align: center;
}
.add-review-container .form_element {
  margin-top: 20px;
}
.add-review-container .title {
  font-size: 33px;
  font-weight: 300;
  color: #02bcd4;
}
.add-review-container .form-input {
  font-size: 20px;
  border-radius: 3px;
  border: 1px solid #ebebeb;
  box-sizing: border-box;
  padding: 11px 31px;
  background: #f7f7f7;
  font-weight: 300;
  width: 80%;
}
.add-review-container .label {
  display: block;
  text-align: left;
  margin-left: 10%;
  color: #02bcd4;
  font-size: 21px;
  margin-bottom: 3px;
}
.add-review-container textarea {
  background: #f7f7f7;
  border: 1px solid #ebebeb;
  border-radius: 3px;
  width: 80%;
  padding: 20px;
  box-sizing: border-box;
  font-size: 14px;
}
.add-review-container .add-button {
  cursor: pointer;
  font-size: 20px;
  padding: 10px 52px;
  background: #02bcd4;
  color: #fff;
  font-weight: 300;
  border-radius: 3px;
  border: none;
  margin-top: 50px;
  margin-bottom: 50px;
  width: 80%;
}
.add-review-container .add-button:hover {
  transform: scale(1.05);
  background: #009eb3;
}
.add-review-container .add-button:active {
  transform: scale(1.01);
  background: #0495a8;
}
.add-review-container .add-button:disabled {
  cursor: not-allowed;
  transform: none;
  background: grey;
}
.add-review-container .error {
  color: #ff5722;
  margin-top: 5px;
}
.add-review-container .editor-wrapper {
  width: 80%;
  margin-left: 10%;
  border: 1px solid #ebebeb;
  background-color: #f7f7f7;
}
.add-review-container .rdw-emoji-modal {
  left: -55px;
}
.add-review-container .rdw-link-modal {
  left: -100px;
}
.add-review-container .rdw-image-modal {
  left: -10px;
}
.add-review-container .editor-self {
  padding-left: 10px;
  padding-right: 10px;
}
.add-review-container .field-error {
  border: 2px solid #ff5722;
  color: #ff5722;
}
