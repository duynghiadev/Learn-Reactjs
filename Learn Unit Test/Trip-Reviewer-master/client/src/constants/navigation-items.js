import * as routes from './routes';

export const home = 'HOME';
export const userProfile = 'USER_PROFILE';
export const signup = 'SIGNUP';
export const login = 'LOGIN';
export const userReviews = 'USER_REVIEWS';
export const addTrip = 'ADD_TRIP';
export const logout = 'LOGOUT';

const items = {
  [home]: {
    id: 1,
    icon: 'home',
    text: 'Home',
    link: routes.HOME,
    restricted: false,
  },
  [userProfile]: {
    id: 2,
    icon: 'user',
    text: 'Profile',
    link: routes.USER_PROFILE,
    restricted: true,
  },
  [signup]: {
    id: 3,
    icon: 'user-plus',
    text: 'Signup',
    link: routes.REGISTER,
    restricted: false,
    exclude: true,
  },
  [login]: {
    id: 4,
    icon: 'sign-in-alt',
    text: 'Login',
    link: routes.LOGIN,
    restricted: false,
    exclude: true,
  },
  [userReviews]: {
    id: 5,
    icon: 'edit',
    text: 'Edit',
    link: routes.USER_REVIEWS,
    restricted: true,
  },
  [addTrip]: {
    id: 6,
    icon: 'file-alt',
    text: 'Add',
    link: routes.ADD_TRIP,
    restricted: true,
  },
  [logout]: {
    id: 7,
    icon: 'sign-out-alt',
    text: 'Logout',
    link: routes.LOGOUT,
    restricted: true,
  },
};

export default items;
