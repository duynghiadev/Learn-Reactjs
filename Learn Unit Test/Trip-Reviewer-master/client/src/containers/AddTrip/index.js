import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Editor } from 'react-draft-wysiwyg';
import { EditorState, convertToRaw } from 'draft-js';
import 'react-draft-wysiwyg/dist/react-draft-wysiwyg.css';

import StarsRating from '../../components/StarsRating';
import CountrySelector from '../../components/CountrySelector';
import { addTrip, clearNewTrip } from '../../actions';
import { TRIPS } from '../../constants/routes';
import toolbar from '../../constants/toolbar';

import './styles.css';

class AddTrip extends Component {
  state = {
    editorState: EditorState.createEmpty(),
    formdata: {
      title: '',
      country: 'Spain',
      description: '',
      duration: '',
      rating: 0,
      expences: '',
    },
    hideError: {
      title: false,
      country: false,
      description: false,
      duration: false,
      rating: false,
      expences: false,
    },
    loading: false,
  };

  componentWillReceiveProps(newProps) {
    const { newtrip } = newProps.trips;

    if (newtrip && newtrip.success) {
      newProps.history.push(`${TRIPS}/${newtrip.tripId}`);
    } else {
      this.setState({
        loading: false,
        hideError: {
          title: false,
          country: false,
          description: false,
          duration: false,
          rating: false,
          expences: false,
        },
      });
    }
  }

  componentWillUnmount() {
    this.props.dispatch(clearNewTrip());
  }

  onEditorStateChange = editorState => {
    const newFormdata = { ...this.state.formdata };
    const newHideError = {
      ...this.state.hideError,
      description: true,
    };
    const contentState = editorState.getCurrentContent();
    const rawState = convertToRaw(contentState);

    newFormdata.description = JSON.stringify(rawState);

    this.setState({
      editorState,
      formdata: newFormdata,
      hideError: newHideError,
    });
  };

  getErrorClass(fieldName) {
    return this.formFieldHasError(fieldName) ? 'field-error' : '';
  }

  formFieldHasError(fieldName) {
    const { newtrip } = this.props.trips;
    const { hideError } = this.state;
    const errors = newtrip && newtrip.error && newtrip.error.errors;

    return errors && errors[fieldName] && !hideError[fieldName];
  }

  handleInput = event => {
    const newFormdata = { ...this.state.formdata };
    const newHideError = { ...this.state.hideError };
    const { value, name } = event.target;

    newFormdata[name] = value;
    newHideError[name] = true;

    this.setState({
      formdata: newFormdata,
      hideError: newHideError,
    });
  };

  handleCountrychange = countryObject => {
    const newFormdata = {
      ...this.state.formdata,
      country: countryObject.countryName,
    };

    this.setState({
      formdata: newFormdata,
    });
  };

  handleRating = rating => {
    const newFormdata = {
      ...this.state.formdata,
      rating,
    };
    const newHideError = {
      ...this.state.hideError,
      rating: true,
    };

    this.setState({
      formdata: newFormdata,
      hideError: newHideError,
    });
  };

  submitForm = event => {
    event.preventDefault();
    this.props.dispatch(
      addTrip({
        ...this.state.formdata,
        ownerId: this.props.users.login.id,
      })
    );
    this.setState({ loading: true });
  };

  renderError(fieldName) {
    if (this.formFieldHasError(fieldName)) {
      const error = this.props.trips.newtrip.error.errors[fieldName];

      return <div className="error">{error.message}</div>;
    }
    return null;
  }

  render() {
    const { loading, formdata } = this.state;
    const { title, country, duration, rating, expences } = formdata;

    return (
      <div className="add-review-container limited-width">
        <form onSubmit={this.submitForm}>
          <h2 className="title">Add a story</h2>

          <div className="form_element">
            <span className="label">Title:</span>
            <input
              type="text"
              name="title"
              className={`form-input ${this.getErrorClass('title')}`}
              placeholder="Enter title"
              value={title}
              onChange={this.handleInput}
            />
          </div>
          {this.renderError('title')}

          <div className="form_element">
            <span className="label">Trip to:</span>
            <CountrySelector
              defaultCountry={country}
              getSelectedCountry={this.handleCountrychange}
            />
          </div>

          <div className="form_element">
            <span className="label">Description:</span>
            <Editor
              editorState={this.state.editorState}
              wrapperClassName={`editor-wrapper ${this.getErrorClass('description')}`}
              editorClassName="editor-self"
              onEditorStateChange={this.onEditorStateChange}
              toolbar={toolbar}
            />
          </div>
          {this.renderError('description')}

          <div className="form_element">
            <span className="label">Duration (days):</span>
            <input
              type="number"
              name="duration"
              className={`form-input ${this.getErrorClass('duration')}`}
              placeholder="Enter duration"
              value={duration}
              onChange={this.handleInput}
            />
          </div>
          {this.renderError('duration')}

          <div className="form_element">
            <span className="label">Expenses ($):</span>
            <input
              type="number"
              name="expences"
              className={`form-input ${this.getErrorClass('expences')}`}
              placeholder="Enter expenses"
              value={expences}
              onChange={this.handleInput}
            />
          </div>
          {this.renderError('expences')}

          <div className="form_element">
            <StarsRating
              rating={rating}
              error={this.getErrorClass('rating')}
              onChange={this.handleRating}
            />
          </div>
          {this.renderError('rating')}

          <button type="submit" className="add-button" disabled={loading}>
            Add story
          </button>
        </form>
      </div>
    );
  }
}

AddTrip.propTypes = {
  dispatch: PropTypes.func.isRequired,
  users: PropTypes.object.isRequired,
  trips: PropTypes.object.isRequired,
};

function mapStateToProps(state) {
  return {
    trips: state.trips,
  };
}

export default connect(mapStateToProps)(AddTrip);
