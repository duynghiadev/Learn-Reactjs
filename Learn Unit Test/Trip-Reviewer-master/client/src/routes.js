import React from 'react';
import { Switch, Route } from 'react-router-dom';

import NotFound from './components/404';
import Home from './containers/Home';
import User from './components/User';
import EditUserProfile from './containers/EditUserProfile';
import Logout from './components/Logout';
import Login from './containers/Login';
import ResetPass from './containers/ResetPass';
import Register from './containers/Register';
import AddTrip from './containers/AddTrip';
import EditTrip from './containers/EditTrip';
import TripView from './containers/TripView';
import UserReviews from './containers/UserReviews';

import Layout from './hoc/layout';
import Auth from './hoc/auth';

import * as routes from './constants/routes';

const Routes = () => (
  <Layout>
    <Switch>
      <Route path={routes.HOME} exact component={Auth(Home, null)} />
      <Route path={routes.LOGIN} exact component={Auth(Login, false)} />
      <Route path={routes.RESET} exact component={Auth(ResetPass, false)} />
      <Route path={routes.LOGOUT} exact component={Auth(Logout, true)} />
      <Route path={routes.REGISTER} exact component={Auth(Register, false)} />
      <Route path={routes.USER_PROFILE} exact component={Auth(User, true)} />
      <Route path={routes.USER_REVIEWS} exact component={Auth(UserReviews, true)} />
      <Route path={`${routes.EDIT_PROFILE}`} exact component={Auth(EditUserProfile, true)} />
      <Route path={`${routes.EDIT_POST}/:id`} exact component={Auth(EditTrip, true)} />
      <Route path={routes.ADD_TRIP} exact component={Auth(AddTrip, true)} />
      <Route path={`${routes.TRIPS}/:id`} exact component={Auth(TripView, null)} />
      <Route path="*" component={Auth(NotFound, null)} />
    </Switch>
  </Layout>
);

export default Routes;
