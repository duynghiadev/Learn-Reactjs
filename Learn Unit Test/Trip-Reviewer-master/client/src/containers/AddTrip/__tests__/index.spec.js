import React from 'react';
import configureStore from 'redux-mock-store';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

import AddTrip from '..';
import { addTrip, clearNewTrip } from '../../../actions';
import { TRIPS } from '../../../constants/routes';

jest.mock('react-draft-wysiwyg', () => ({ Editor: 'Editor' }));
jest.mock('draft-js', () => ({
  EditorState: {
    createEmpty: () => ({}),
  },
  convertToRaw: () => ({}),
}));
jest.mock('./../../../components/StarsRating', () => 'StarsRating');
jest.mock('./../../../components/CountrySelector', () => 'CountrySelector');
jest.mock('../../../constants/toolbar', () => ({}));
jest.mock('./../../../actions', () => ({
  addTrip: jest.fn(() => ({
    type: 'test',
  })),
  clearNewTrip: jest.fn(() => ({
    type: 'test',
  })),
}));

const mockStore = configureStore();

const mockComponent = (initialState = {}, props) => {
  const store = mockStore(initialState);

  return <AddTrip {...props} store={store} />;
};

describe('<AddTrip />', () => {
  it('should render component', () => {
    const initialState = {
      trips: {},
    };
    const tree = create(mockComponent(initialState)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with newTrip and success true', () => {
    const initialState = {
      trips: {
        newtrip: {
          success: true,
          tripId: 'test',
        },
      },
    };
    const tree = create(mockComponent(initialState)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with newTrip and errors', () => {
    const initialState = {
      trips: {
        newtrip: {
          success: false,
          error: {
            errors: {
              title: { message: 'title error' },
              description: { message: 'description error' },
              duration: { message: 'duration error' },
              rating: { message: 'rating error' },
              expences: { message: 'expences error' },
            },
          },
        },
      },
    };
    const tree = create(mockComponent(initialState)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('componentWillReceiveProps should redirect when newtrip exists', () => {
    const push = jest.fn();
    const tripId = 'test';
    const path = `${TRIPS}/${tripId}`;
    const initialState = {
      trips: {},
    };
    const props = {
      users: {
        login: {
          id: 'id',
        },
      },
    };
    const newProps = {
      trips: {
        newtrip: {
          success: true,
          tripId,
        },
      },
      history: {
        push,
      },
    };

    const instance = shallow(mockComponent(initialState, props)).dive().instance();

    instance.componentWillReceiveProps(newProps);

    expect(push).toHaveBeenCalledWith(path);
  });

  it('componentWillReceiveProps should not redirect when newtrip does not exists', () => {
    const push = jest.fn();
    const initialState = {
      trips: {},
    };
    const props = {
      users: {
        login: {
          id: 'id',
        },
      },
    };
    const newProps = {
      trips: {
        newtrip: null,
      },
      history: {
        push,
      },
    };

    const instance = shallow(mockComponent(initialState, props)).dive().instance();

    instance.componentWillReceiveProps(newProps);

    expect(push).not.toHaveBeenCalled();
  });

  it('handleInput should change state correctly', () => {
    const initialState = {
      trips: {},
    };
    const event = {
      target: {
        value: 'test',
        name: 'title',
      },
    };

    const instance = shallow(mockComponent(initialState)).dive().instance();

    instance.handleInput(event);

    expect(instance.state.formdata.title).toEqual('test');
  });

  it('onEditorStateChange should change state correctly', () => {
    const initialState = {
      trips: {},
    };
    const props = {
      match: {
        params: {
          id: 'id',
        },
      },
    };
    const editorState = {
      getCurrentContent: () => ({}),
      test: 'test',
    };

    const instance = shallow(mockComponent(initialState, props)).dive().instance();

    instance.onEditorStateChange(editorState);

    expect(instance.state.editorState).toEqual(editorState);
    expect(instance.state.formdata.description).toEqual('{}');
  });

  it('handleRating should change state correctly', () => {
    const initialState = {
      trips: {},
    };
    const rating = 3;

    const instance = shallow(mockComponent(initialState)).dive().instance();

    instance.handleRating(rating);

    expect(instance.state.formdata.rating).toEqual(rating);
  });

  it('handleCountrychange should change state correctly', () => {
    const initialState = {
      trips: {},
    };
    const countryName = 'countryName';
    const coutryObject = { countryName };

    const instance = shallow(mockComponent(initialState)).dive().instance();

    instance.handleCountrychange(coutryObject);

    expect(instance.state.formdata.country).toEqual(countryName);
  });

  it('should dispatch addTrip when submitForm is called', () => {
    const preventDefault = jest.fn();
    const props = {
      users: {
        login: {
          id: 'id',
        },
      },
    };
    const state = {
      country: 'country',
      duration: 'duration',
      ownerId: 'id',
      expences: 'expences',
      rating: 'rating',
      description: 'description',
      title: 'title',
    };
    const initialState = {
      trips: {},
    };
    const event = { preventDefault };

    const instance = shallow(mockComponent(initialState, props)).dive().instance();
    instance.setState({ formdata: state });

    instance.submitForm(event);

    expect(preventDefault).toHaveBeenCalled();
    expect(addTrip).toHaveBeenCalledWith(state);
  });

  it('should dispatch clearNewTrip when componentWillUnmount is called', () => {
    const initialState = {
      trips: {},
    };

    const instance = shallow(mockComponent(initialState)).dive().instance();

    instance.componentWillUnmount();

    expect(clearNewTrip).toHaveBeenCalled();
  });
});
