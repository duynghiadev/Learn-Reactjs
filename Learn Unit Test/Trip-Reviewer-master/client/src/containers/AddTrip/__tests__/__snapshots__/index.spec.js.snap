// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTrip /> should render component 1`] = `
<div
  className="add-review-container limited-width"
>
  <form
    onSubmit={[Function]}
  >
    <h2
      className="title"
    >
      Add a story
    </h2>
    <div
      className="form_element"
    >
      <span
        className="label"
      >
        Title:
      </span>
      <input
        className="form-input "
        name="title"
        onChange={[Function]}
        placeholder="Enter title"
        type="text"
        value=""
      />
    </div>
    <div
      className="form_element"
    >
      <span
        className="label"
      >
        Trip to:
      </span>
      <CountrySelector
        defaultCountry="Spain"
        getSelectedCountry={[Function]}
      />
    </div>
    <div
      className="form_element"
    >
      <span
        className="label"
      >
        Description:
      </span>
      <Editor
        editorClassName="editor-self"
        editorState={Object {}}
        onEditorStateChange={[Function]}
        toolbar={Object {}}
        wrapperClassName="editor-wrapper "
      />
    </div>
    <div
      className="form_element"
    >
      <span
        className="label"
      >
        Duration (days):
      </span>
      <input
        className="form-input "
        name="duration"
        onChange={[Function]}
        placeholder="Enter duration"
        type="number"
        value=""
      />
    </div>
    <div
      className="form_element"
    >
      <span
        className="label"
      >
        Expenses ($):
      </span>
      <input
        className="form-input "
        name="expences"
        onChange={[Function]}
        placeholder="Enter expenses"
        type="number"
        value=""
      />
    </div>
    <div
      className="form_element"
    >
      <StarsRating
        error=""
        onChange={[Function]}
        rating={0}
      />
    </div>
    <button
      className="add-button"
      disabled={false}
      type="submit"
    >
      Add story
    </button>
  </form>
</div>
`;

exports[`<AddTrip /> should render component with newTrip and errors 1`] = `
<div
  className="add-review-container limited-width"
>
  <form
    onSubmit={[Function]}
  >
    <h2
      className="title"
    >
      Add a story
    </h2>
    <div
      className="form_element"
    >
      <span
        className="label"
      >
        Title:
      </span>
      <input
        className="form-input field-error"
        name="title"
        onChange={[Function]}
        placeholder="Enter title"
        type="text"
        value=""
      />
    </div>
    <div
      className="error"
    >
      title error
    </div>
    <div
      className="form_element"
    >
      <span
        className="label"
      >
        Trip to:
      </span>
      <CountrySelector
        defaultCountry="Spain"
        getSelectedCountry={[Function]}
      />
    </div>
    <div
      className="form_element"
    >
      <span
        className="label"
      >
        Description:
      </span>
      <Editor
        editorClassName="editor-self"
        editorState={Object {}}
        onEditorStateChange={[Function]}
        toolbar={Object {}}
        wrapperClassName="editor-wrapper field-error"
      />
    </div>
    <div
      className="error"
    >
      description error
    </div>
    <div
      className="form_element"
    >
      <span
        className="label"
      >
        Duration (days):
      </span>
      <input
        className="form-input field-error"
        name="duration"
        onChange={[Function]}
        placeholder="Enter duration"
        type="number"
        value=""
      />
    </div>
    <div
      className="error"
    >
      duration error
    </div>
    <div
      className="form_element"
    >
      <span
        className="label"
      >
        Expenses ($):
      </span>
      <input
        className="form-input field-error"
        name="expences"
        onChange={[Function]}
        placeholder="Enter expenses"
        type="number"
        value=""
      />
    </div>
    <div
      className="error"
    >
      expences error
    </div>
    <div
      className="form_element"
    >
      <StarsRating
        error="field-error"
        onChange={[Function]}
        rating={0}
      />
    </div>
    <div
      className="error"
    >
      rating error
    </div>
    <button
      className="add-button"
      disabled={false}
      type="submit"
    >
      Add story
    </button>
  </form>
</div>
`;

exports[`<AddTrip /> should render component with newTrip and success true 1`] = `
<div
  className="add-review-container limited-width"
>
  <form
    onSubmit={[Function]}
  >
    <h2
      className="title"
    >
      Add a story
    </h2>
    <div
      className="form_element"
    >
      <span
        className="label"
      >
        Title:
      </span>
      <input
        className="form-input "
        name="title"
        onChange={[Function]}
        placeholder="Enter title"
        type="text"
        value=""
      />
    </div>
    <div
      className="form_element"
    >
      <span
        className="label"
      >
        Trip to:
      </span>
      <CountrySelector
        defaultCountry="Spain"
        getSelectedCountry={[Function]}
      />
    </div>
    <div
      className="form_element"
    >
      <span
        className="label"
      >
        Description:
      </span>
      <Editor
        editorClassName="editor-self"
        editorState={Object {}}
        onEditorStateChange={[Function]}
        toolbar={Object {}}
        wrapperClassName="editor-wrapper "
      />
    </div>
    <div
      className="form_element"
    >
      <span
        className="label"
      >
        Duration (days):
      </span>
      <input
        className="form-input "
        name="duration"
        onChange={[Function]}
        placeholder="Enter duration"
        type="number"
        value=""
      />
    </div>
    <div
      className="form_element"
    >
      <span
        className="label"
      >
        Expenses ($):
      </span>
      <input
        className="form-input "
        name="expences"
        onChange={[Function]}
        placeholder="Enter expenses"
        type="number"
        value=""
      />
    </div>
    <div
      className="form_element"
    >
      <StarsRating
        error=""
        onChange={[Function]}
        rating={0}
      />
    </div>
    <button
      className="add-button"
      disabled={false}
      type="submit"
    >
      Add story
    </button>
  </form>
</div>
`;
